src/rust/catnap/mod.rs

```diff
@@ -235,23 +235,11 @@ impl CatnapLibOS {
     }
 
     /// Sets a socket as a passive one.
-    pub fn listen(&mut self, qd: QDesc, mut backlog: usize) -> Result<(), Fail> {
+    pub fn listen(&mut self, qd: QDesc, backlog: usize) -> Result<(), Fail> {
         trace!("listen() qd={:?}, backlog={:?}", qd, backlog);
 
-        // Truncate backlog length.
-        if backlog > libc::SOMAXCONN as usize {
-            let cause: String = format!(
-                "backlog length is too large, truncating (qd={:?}, backlog={:?})",
-                qd, backlog
-            );
-            debug!("listen(): {}", &cause);
-            backlog = libc::SOMAXCONN as usize;
-        }
-
-        // Round up backlog length.
-        if backlog == 0 {
-            backlog = 1;
-        }
+        // We just assert backlog here, because it was previously checked at PDPIX layer.
+        debug_assert!((backlog > 0) && (backlog <= libc::SOMAXCONN as usize));
```

Comment:
Is unreachable more appropriate here?

Reply:
I don't think so. If you are referring to:

```rust
if (backlog > 0) && (backlog <= libc::SOMAXCONN as usize) {
    unreachable("bl abla");
}
```

This will introduce an extra comparison, even on release mode.